tools/myst/enc/enc.c

```diff
@@ -420,6 +449,10 @@ static uint64_t _vectored_handler(oe_exception_record_t* er)
         }
     }
 
+    /* Restore FS if needed */
+    if (saved_fs)
+        asm volatile("wrfsbase %0" ::"r"(saved_fs));
+
     return _forward_exception_as_signal_to_kernel(er);
```

Comment:
Within a signal handler the FS value is that of the application's. C++ allows certain thread-local variables to be accessed from a signal handler. This means that the FS value ought to be applications within the handler - neither OE's nor mystikos kernel's.

Reply:
The kernel code will switch the FS to user value before calling the user handler: https://github.com/deislabs/mystikos/blob/178bea1b744799cddb365e9f5a8cb6c9e625bb32/kernel/signal.c#L559

Reply:
One thing I noticed was that if an application changes FS from appFS to newFS and then a signal happens, then in Linux the value of FS within the signal handler is newFS. In mystikos, the value within the handler is appFS. 
I am not sure if this will cause problems; but it would be good to confirm the Linux behavior and have Mystikos match it so.